File: University Projects/C# Projects/Design and Programming Environments/Proiect/services/IService.cs
using model;
using System;
using System.Collections.Generic;
using System.Text;

namespace services
{
    public interface IService
    {
        void Login(string username, string password, IObserver client);
        void Logout(string username, IObserver client);

        public List<SwimmingTestDTO> GetAllSwimmingTestDTOs();
        public List<SwimmingTest> GetAllSwimmingTestsForParticipant(int participantId);
        public List<ParticipantDTO> GetAllParticipantDTOs(int swimmingTestId);

        public void AddParticipant(String name, int age, List<SwimmingTest> swimmingTests);
        public void UpdateParticipant(ParticipantDTO participantDTO, List<SwimmingTest> swimmingTests);
    }
}

File: University Projects/Java Projects/Design and Programming Environments/Proiect/Services/src/main/java/app/services/IService.java
package app.services;

import java.util.List;
import app.model.Participant;
import app.model.ParticipantDTO;
import app.model.SwimmingTest;
import app.model.SwimmingTestDTO;

public interface IService {
    void login(String username, String password, IObserver client) throws AppException;
    void logout(String username, IObserver client) throws AppException;

    List<SwimmingTestDTO> getAllSwimmingTestDTOs() throws AppException;
    List<SwimmingTest> getSwimmingTestsForParticipant(Integer participantId) throws AppException;
    List<ParticipantDTO> getAllParticipantDTOs(int swimmingTestId) throws AppException;

    void addParticipant(String name, int age, List<SwimmingTest> swimmingTests) throws AppException;
    void updateParticipant(ParticipantDTO participantDTO, List<SwimmingTest> swimmingTests) throws AppException;
}
